data-generation/src/main/scala/thesis/TopologyGraphGenerator.scala

```diff
@@ -5,19 +5,45 @@ import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.expressions.Window
 import org.apache.spark.sql.functions.{col, lag, last, when}
 import org.apache.spark.sql.{SparkSession, functions}
+import org.slf4j.LoggerFactory
 
 import java.time.Instant
 
-final case class TimeInterval(start:Instant, stop:Instant)
+final case class TimeInterval(start: Instant, stop: Instant)
+
+sealed abstract class DataSource
+
+object DataSource {
+  final case class Reptilian() extends DataSource
+
+  final case class FbMessages() extends DataSource
```

Comment:
```suggestion
  final case object Reptilian extends DataSource

  final case object FbMessages extends DataSource
```
Er det bedre å bruke case objects her? Kanskje noe mer intuitivt når man ikke har noen parametre

Reply:
Bra catcha! Var ikke kjent med at case objects også var en greie